### ISSUE TITLE
Authenticate hook ('display name' of user)

### ISSUE THREAD

**Issue by cweiss1** [ether/etherpad-lite#5065]
**Describe the bug**
I am writing a plugin with the authenticate and authorize hook (to work with another system auth flow). The hook works fine - users are allowed or not based on particular criteria. HOWEVER, when a user gets into the pad - the 'Display Name' is always blank!!. 
I looked at the example in the doc - and followed that for the username - what I am doing, for example:
     username = 'Some User";
     users[username].username = username;
     //tried setting users[username].name = username;  -- does not help

any help appreciated - I must be missing something obvious to set?

**To Reproduce**
Follow the example - https://github.com/ether/etherpad-lite/wiki/Creating-a-plugin - for authenticate.
Set users[username].name="Some User"

**Expected behavior**
"Some User" is displayed when the user clicks on his person icon or in chats.

**Server (please complete the following information):**
 - Etherpad version: https://hub.docker.com/r/etherpad/etherpad
 - OS: [e.g., Ubuntu 20.04] - centos 7
 - Node.js version (`node --version`): (what is in the etherpad docker - 10.17 I think?)
 - npm version (`npm --version`): (same as above)

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] mac
 - Browser [e.g. chrome, safari] firefox

**Comment by rhansen** [comment]
This is how ep_openid_connect handles it:  https://github.com/ether/ep_openid_connect/blob/8409f583b1714497e595676df3f8b2df8b863bb0/index.js#L113-L130

**Comment by cweiss1** [comment]
@rhansen thanks for that pointer. But from what I read - the oidc auth (handleMessage) is RESETTING the displayName to what the token has - since I guess they don't want to allow users to update the name. (am I mistaken?). 
One thing I did notice is -   https://github.com/ether/ep_openid_connect/blob/8409f583b1714497e595676df3f8b2df8b863bb0/index.js#L99 let me try that - I think I was using displayName instead displayname for some reason - doh!

**Comment by cweiss1** [comment]
no soup for me!
tried setting :
users["Some User"] to {
  name: 'Some User',
  username: 'Some User',
  displayname: 'Some User'
}

So unless I am supposed to set and reset it all the time?

**Comment by cweiss1** [comment]
solved! - @rhansen thanks for the pointer - but it seems handleMessage (client-ready) might be what 'fixes' the displayname not showing up issue - I added that to my code and now see the displayname.
It is a bit weird how the doc does not talk about this - or perhaps it did and its buried somewhere? (the handlemessage and the need to set the displayname AFTER I have already provided the entire user creds in authenticate seems a bit odd - but it works for now - so I will close this issue)

Thanks for your help!
